.github/actions/git_checkout_from_cache/action.yml

```diff
@@ -0,0 +1,34 @@
+name: git_checkout_from_cache
+description: Checkout git from cache
+runs:
+  using: composite
+  steps:
+  - name: restore_cache
+    uses: actions/cache@v3.3.1
+    with:
+      key: source-v{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-{{ .Revision }}
+      path: UPDATE_ME
+      restore-keys: |-
+        source-v{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-{{ .Revision }}
+        source-v{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-
+        source-v{{ .Environment.CACHE_VERSION }}-
+  - name: Fetch git tags
+    run: |-
+      mkdir -p ~/.ssh
+      ssh-keyscan github.com >> ~/.ssh/known_hosts
+      echo "github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==" >> ~/.ssh/known_hosts
+      # Fetch tags if git cache is present
+      if [ -e /home/circleci/project/.git ]
+      then
+        git fetch origin --tags
+      fi
+    shell: bash
+  - uses: actions/checkout@v3.6.0
+  # - name: Compress git objects
+  #   run: git gc
```

Comment:
this is for git garbage collection which didn't work. need to check it more, but for now even without it everything work fine.